scipy/stats/stats.py

```diff
@@ -6110,20 +6155,85 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
                                  alternative=alternative)
 
     else:
-        v1 = np.var(a, axis, ddof=1)
-        v2 = np.var(b, axis, ddof=1)
         n1 = a.shape[axis]
         n2 = b.shape[axis]
 
+        if trim == 0:
+            v1 = np.var(a, axis, ddof=1)
+            v2 = np.var(b, axis, ddof=1)
+            m1 = np.mean(a, axis)
+            m2 = np.mean(b, axis)
+        else:
+            #
+            v1, m1, n1 = _ttest_trim_var_mean_len(a, trim, n1, axis)
+            v2, m2, n2 = _ttest_trim_var_mean_len(b, trim, n2, axis)
+
         if equal_var:
             df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
         else:
             df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
-        res = _ttest_ind_from_stats(np.mean(a, axis), np.mean(b, axis),
-                                    denom, df, alternative)
+
+        res = _ttest_ind_from_stats(m1, m2, denom, df, alternative)
     return Ttest_indResult(*res)
 
 
+def _ttest_trim_var_mean_len(a, trim, n, axis):
+    """Variance, mean, and length of winsorized input along specified axis"""
+    # for use with `ttest_ind` when trimming.
+    # further calculations in this test assume that the inputs are sorted.
+    # From [4] Section 1 "Let x_1, ..., x_n be n ordered observations..."
+    a = np.sort(a, axis=axis)
+
+    # `g_*` is the number of elements to be replaced on each tail, converted
+    # from a percentage amount of trimming
+    g = int(n * trim)
+
+    # Calculate the Winsorized variance of the input samples according to
+    # specified `g`
+    v = _calculate_winsorized_variance(a, g, n, axis)
+
+    # the total number of elements in the trimmed samples
+    n -= 2 * g
+
+    # calculate the g-times trimmed mean, as defined in [4] (1-1)
+    m = trim_mean(a, trim, axis=axis)
+    return v, m, n
+
+
+def _calculate_winsorized_variance(a, g, n, axis):
```

Comment:
```suggestion
def _calculate_winsorized_variance(a, g, axis):
```
No need to pass this in. If you need it in here, get it from `a.shape[axis]`, but I think you don't need it - you can just use negative indices.

Reply:
I originally included it to cover the case where `g=0`, since a[-0] is the front of the array when I wanted the end. I will use a branch instead for if g=0 to avoid it.

Reply:
Interesting. Hadn't thought about that. Good to short circuit for `g==0`.